File: adapters/tam/src/rftprequest.cpp
/******************************************************************
* Copyright (c) DuckDigit Technologies, Inc.  ALL RIGHTS RESERVED
* $Header: /home/cvsroot/ai/adapters/tam/src/rftprequest.cpp,v 1.1 2009/09/20 07:15:31 secwind Exp $
*****************************************************************/
#include <dbclient.h>

/**
 * The constructor extracts out the elements of the request from 
 * incoming byte array.  The string pointers are no longer valid 
 * after the buf is changed so don't rely after process is called.
 */
RftpRequest_::RftpRequest_ (char *buf, char *filename)
	: DbRequest_ (buf, FTP)
{
	appendString (filename);
}

File: adapters/infinity/java/com/duckdigit/isam/RftpRequest.java
/*==================================================================================================
* Copyright (c) DuckDigit Technologies, Inc.  2008,2009
* ALL RIGHTS RESERVED
* $Header: /home/cvsroot/ai/adapters/infinity/java/com/duckdigit/isam/RftpRequest.java,v 1.1 2010/10/19 00:44:26 secwind Exp $
==================================================================================================*/
package com.duckdigit.isam;

import java.util.*;

public class RftpRequest extends IsamRequest
{
	public RftpRequest (byte buf [], String path, boolean download)
		throws IsamException
	{
		super (buf, IsamRequest.FTP);
		this.appendString (path);
		this.appendInt (download ? 1 : 0);
	}
}
